Sources/ArcGISToolkit/Components/Authentication/CertificatePickerViewModifier.swift

```diff
@@ -92,11 +103,31 @@ import ArcGIS
         challenge.resume(with: .cancel)
     }
 
-    private func showCertificateImportError(_ error: Error?) async {
+    private func showCertificateError(_ error: CertificateError) async {
         // This is required to prevent an "already presenting" error.
         try? await Task.sleep(nanoseconds: 100_000)
-        certificateImportError = error as? CertificateImportError
-        showCertificateImportError = true
+        certificateError = error
+        showCertificateError = true
+    }
+}
+
+extension CertificatePickerViewModel.CertificateError {
+    var message: String {
+        switch self {
+        case .couldNotAccessCertificateFile:
+            return "Could not access the certificate file."
+        case .importError(let certificateImportError):
+            switch certificateImportError {
+            case .invalidData:
+                return "The certificate file was invalid."
+            case .invalidPassword:
+                return "The password was invalid."
+            default:
+                return "The certificate file or password was invalid."
+            }
```

Comment:
I think actually I would delegate this to the localized description or message of the `CertificateImportError`.

Reply:
You mean just use `localizedDescription` property of the error or adopt, `LocalizedError` on `CertificateImportError` and move these strings in it? If we just use the `localizedDescription` then the value will be something like this, `"The operation couldn’t be completed. (ArcGIS.CertificateImportError error -26275.)"`

Reply:
We could do the `LocalizedError` thing.  it will have to be done at some point when we localize the toolkit. Or you can just implement a `message` property on CertificateIMportError.  Either way would be preferable to the embedded `switch`.  Might be best to just do the right thing (`LocalizedError`) now while we are editing the code.

Reply:
done